server/jetstream_cluster.go

```diff
@@ -4281,6 +4288,11 @@ func (cc *jetStreamCluster) selectPeerGroup(r int, cluster string, cfg *StreamCo
 	// Sort based on available from most to least.
 	sort.Slice(nodes, func(i, j int) bool { return nodes[i].avail > nodes[j].avail })
 
+	// If we are placing a replicated stream, let's sort based in haAssets, as that is more important to balance.
+	if cfg.Replicas > 1 {
+		sort.SliceStable(nodes, func(i, j int) bool { return nodes[i].ha < nodes[j].ha })
```

Comment:
one sort step? 

If nodes[i].avail == nodes[i].avail {
    return nodes[i].ha < nodes[j].ha
} else {
    return nodes[i].avail > nodes[j].avail
}

Reply:
Right now I am only sorting and preferring HA when placing a replicated stream. These are are wittled down to a cluster, so think in terms of 3 or 5 nodes in the general case.